src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -0,0 +1,965 @@
+            WriteVerbose(reqVerboseMsg);
+        }
+
+        private void ReportResponseStatus(HttpResponseMessage responseMessage)
+        {
+            var contentType = ContentHelper.GetContentType(responseMessage);
+            var respVerboseMsg = StringFormatCurrentCulture("received {0}-byte response of content type {1}",
+                responseMessage.Content.Headers.ContentLength,
+                contentType);
+            WriteVerbose(respVerboseMsg);
+        }
+
+        private static string StringFormatCurrentCulture(string format, params object[] args)
+        {
+            return string.Format(CultureInfo.CurrentCulture, format, args);
+        }
+
+        private static string FormatDictionary(IDictionary content)
+        {
+            if (content == null)
+                throw new ArgumentNullException(nameof(content));
+
+            var bodyBuilder = new StringBuilder();
+            foreach (string key in content.Keys)
+            {
+                if (0 < bodyBuilder.Length) bodyBuilder.Append("&");
+
+                var value = content[key];
+
+                // URLEncode the key and value
+                var encodedKey = WebUtility.UrlEncode(key);
+                var encodedValue = string.Empty;
+                if (value != null) encodedValue = WebUtility.UrlEncode(value.ToString());
+
+                bodyBuilder.AppendFormat("{0}={1}", encodedKey, encodedValue);
+            }
+
+            return bodyBuilder.ToString();
+        }
+
+        private HttpRequestMessage GetRequest(Uri uri)
+        {
+            var requestUri = PrepareUri(uri);
+            var httpMethod = GetHttpMethod(Method);
+            // create the base WebRequest object
+            var request = new HttpRequestMessage(httpMethod, requestUri);
+
+            // pull in session data
+            if (GraphRequestSession.Headers.Count > 0)
+            {
+                GraphRequestSession.ContentHeaders.Clear();
+                foreach (var entry in GraphRequestSession.Headers)
+                {
+                    if (HttpKnownHeaderNames.ContentHeaders.Contains(entry.Key))
+                    {
+                        GraphRequestSession.ContentHeaders.Add(entry.Key, entry.Value);
+                    }
+                    else
+                    {
+                        if (SkipHeaderValidation)
+                        {
+                            request.Headers.TryAddWithoutValidation(entry.Key, entry.Value);
+                        }
+                        else
+                        {
+                            request.Headers.Add(entry.Key, entry.Value);
+                        }
+                    }
+                }
+            }
+
+            // Set 'Transfer-Encoding: chunked' if 'Transfer-Encoding' is specified
+            if (GraphRequestSession.Headers.ContainsKey(HttpKnownHeaderNames.TransferEncoding))
+            {
+                request.Headers.TransferEncodingChunked = true;
```

Comment:
What happens if the header is set to `false` explicitly?

Reply:
It cannot be set to false, since that's now an allowed value.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding
Allowed Values are:- 

1. - Transfer-Encoding: chunked
2. - Transfer-Encoding: compress
3. - Transfer-Encoding: deflate
4. - Transfer-Encoding: gzip
5. - Transfer-Encoding: identity

The property request.Headers.TransferEncodingChunked, just causes (1) to be emitted.